src/main/java/net/minecraftforge/dimension/DynamicDimensionManager.java

```diff
@@ -0,0 +1,221 @@
+	{
+		Validate.notNull(server, "Must provide server when creating world");
+		Validate.notNull(loc, "Must provide valid ResourceLocation");
+		
+		RegistryKey<Dimension> registryKey = RegistryKey.func_240903_a_(Registry.field_239700_af_, loc);
+		RegistryKey<World> registryKey1 = RegistryKey.func_240903_a_(Registry.field_239699_ae_, registryKey.func_240901_a_());
+		
+		Dimension dimension = this.serverWorldInfo.func_230418_z_().func_236224_e_().func_230516_a_(registryKey);
+		
+        DimensionType type = dimension.func_236063_b_();
+        ChunkGenerator gen = dimension.func_236064_c_();
+        
+        server.loadDimension(registryKey1, type, gen);
+	}
+	
+	/**
+	 * Retrieves the dimension type under the specified {@link ResourceLocation}
+	 * @param loc the location of the dimension
+	 * @return the dimension type or null
+	 */
+	@Nullable
+	public DimensionType getDimensionType(ResourceLocation loc) 
+	
+	{
+		Validate.notNull(loc, "Must provide valid ResourceLocation");
+		
+		RegistryKey<DimensionType> key = RegistryKey.func_240903_a_(Registry.field_239698_ad_, loc);
+		
+		Registry<DimensionType> reg = this.dynamicRegistries.func_230520_a_();
+		DimensionType type = reg.func_230516_a_(key);
+		
+		return type;
+	}
+	
+	/**
+	 * Retrieves the dimension under the specified {@link ResourceLocation}
+	 * @param loc the location of the dimension
+	 * @return the dimension
+	 */
+	@Nullable
+	public Dimension getDimension(ResourceLocation loc) 
+	{
+		Validate.notNull(loc, "Must provide valid ResourceLocation");
+		
+		RegistryKey<Dimension> registryKey = RegistryKey.func_240903_a_(Registry.field_239700_af_, loc);
+		
+		Dimension dimension = this.serverWorldInfo.func_230418_z_().func_236224_e_().func_230516_a_(registryKey);
+		
+		return dimension;
+	}
+	
+	/**
+	 * Retrieves worlds that are registered under the specified dimension type
+	 * @param server the minecraft server
+	 * @param loc the location of the world
+	 * @return the world
+	 */
+	@Nullable
+	public List<ServerWorld> getServerWorlds(MinecraftServer server, DimensionType type) 
+	{
+		Validate.notNull(server, "Must provide server when creating world");
+		Validate.notNull(type, "Must provide DimensionType");
+		
+		Registry<DimensionType> reg = this.dynamicRegistries.func_243612_b(Registry.field_239698_ad_);
+		
+		List<ServerWorld> worlds = new ArrayList<ServerWorld>();
+		
+		for(ServerWorld world : server.getWorlds()) 
+		{
+			if(reg.getKey(world.func_230315_m_()).equals(reg.getKey(type))) 
+			{
+				worlds.add(world);
+			}
+		}
+		
+		return worlds;
+	}
+	
+	/**
+	 * Retrieves the world instance of the dimension under the specified {@link ResourceLocation}
+	 * @param server the minecraft server
+	 * @param loc the location of the world
+	 * @return the world
+	 */
+	@Nullable
+	public World getServerWorld(MinecraftServer server, ResourceLocation loc) 
+	{
+		Validate.notNull(server, "Must provide server when creating world");
```

Comment:
why "when creating a world"? you say in the doc you retrieve the instance, I would assume this means the instance has already been created.

Reply:
Might be me thinking multiple things at once and then mixing up. Must mean when retrieving world.